core/controllers/story_viewer.py

```diff
@@ -170,6 +171,7 @@ def get(self, story_id, node_id):
     @acl_decorators.can_access_story_viewer_page
     def post(self, story_id, node_id):
         story = story_fetchers.get_story_by_id(story_id)
+        topic = topic_fetchers.get_topic_by_id(story.corresponding_topic_id)
```

Comment:
Ditto deleted topic, the topic completion update should not occur if topic is deleted.

Reply:
Done.